src/toxcore/dht/packet/crypto_data.rs

```diff
@@ -163,6 +163,17 @@ impl ToBytes for CryptoDataPayload {
     }
 }
 
+impl CryptoDataPayload {
+    /// Get data skipping zeros at the front
+    pub fn real_data(&self) -> Vec<u8> {
+        self.data
+            .iter()
+            .cloned()
```

Comment:
Do we really need to clone it once more?

Reply:
Because `iter()` returns iterator over `&u8` and I need just value `u8`. And `cloned()` method just clones inner value from reference.